pkg/edgehub/common/http/http_test.go

```diff
@@ -100,3 +97,60 @@ func TestSendRequest(t *testing.T) {
 	common.AssertIntEqual(t, strconv.Itoa(http.StatusNotFound), strconv.Itoa(resp.StatusCode), "return code not match")
 
 }
+
+//TestNewHTTPSclient() tests the creation of a new HTTPS client with proper values
+func TestNewHTTPSclient(t *testing.T) {
+	err := util.GenerateTestCertificate("/tmp/kubeedge/testData/", "edge", "edge")
+	if err != nil {
+		t.Errorf("Error in generating fake certificates: %v", err)
+	}
+	certificate, err := tls.LoadX509KeyPair(CertFile, KeyFile)
+	if err != nil {
+		t.Errorf("Error in loading key pair: %v", err)
+	}
+	type args struct {
+		certFile string
+		keyFile  string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *http.Client
+		wantErr bool
+	}{
+		{"TestNewHTTPSclient: ", args{
+			keyFile:  KeyFile,
+			certFile: CertFile,
+		}, &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					RootCAs:      x509.NewCertPool(),
+					Certificates: []tls.Certificate{certificate},
+					MinVersion:   tls.VersionTLS12,
+					CipherSuites: []uint16{
+						tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+					},
+					InsecureSkipVerify: true},
+			},
+			Timeout: connectTimeout,
+		}, false},
+
+		{"TestNewHTTPSclient: ", args{
+			keyFile:  "WrongKeyFilePath",
+			certFile: "WrongCertFilePath",
+		}, nil,
+			true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := NewHTTPSclient(tt.args.certFile, tt.args.keyFile)
+			if (err != nil) != tt.wantErr {
```

Comment:
use reflect.Deepequal to check error ?

Reply:
This has been done deliberately as the errors being returned are being returned from inbuilt go files which have error types like os.PathError, which is causing reflect.DeepEqual() to fail in the test case. Hence this solution.